src/conversion/wkb.jl

```diff
@@ -2,6 +2,115 @@
 # Licensed under the MIT License. See LICENSE in the project root.
 # ------------------------------------------------------------------
 
+function meshes2wkb(io, geom)
+  wkbtype = _wkbtype(geom)
+  write(io, htol(one(UInt8)))
+  write(io, htol(UInt32(wkbtype)))
+  geom2wkb(io, wkbtype, geom)
+end
+
+
+function _wkbtype(geometry)
+  if geometry isa Point
+    # wkbPoint 
+    return 1
+  elseif geometry isa Rope || geometry isa Ring
+    # wkbLineString
+    return 2
+  elseif geometry isa PolyArea
+    # wkbPolygon
+    return 3
+  elseif geometry isa Multi
+    # wkbMulti
+    fg = first(parent(geometry))
+    return _wkbtype(fg) + 3
+  end
+end
+
+function geom2wkb(io, wkbtype, geom)
+  if wkbtype > 3
+    multi2wkb(io, wkbtype, geom)
+  else
+    single2wkb(io, wkbtype, geom)
+  end
```

Comment:
Try to follow the same logic of the reader, and avoid this separation between single and multi geometries. You can place all cases into a single if-else paragraph that calls the corresponding function for the wkbtype.